action/builder.go

```diff
@@ -139,3 +143,47 @@ func (b *EnvelopeBuilder) Build() Envelope {
 	}
 	return &b.elp
 }
+
+// BuildStakingAction loads staking action into envelope from abi-encoded data
+func (b *EnvelopeBuilder) BuildStakingAction(data []byte) (Envelope, error) {
```

Comment:
types.Transaction may be easier to understand.

Reply:
ok. `elpBuilder.BuildExecution(tx)`, `elpBuilder.BuildTransfer(tx)`, `elpBuilder.BuildStakingAction(tx)` looks more clear for me